### PR TITLE
Allow specifying a custom image optimizer image cache folder using an environment variable

### PR DESCRIPTION
**Pull request by zmarty**
**Allow specifying a custom image optimizer image cache folder using the optional NEXT_IMAGE_CACHE_PATH environment variable**

Please go easy on me, this is my first PR :) I have validated that this code works by setting the environment variable to a location outside of my project, then running my project against the modified next.js instance in this branch. The cache files get created in the external folder and are served correctly.

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.

I opened a discussion with my proposed implementation a while ago, but nobody replied: https://github.com/vercel/next.js/issues/25733

- [X] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [X] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.

## Documentation / Examples

- [X] Make sure the linting passes
These two linters pass: lint-typescript, lint-eslint

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,3 +37,6 @@ test-timings.json
 # Vercel
 .vercel
 .now
+
```
Comment (Dalimil) [inline .gitignore:40]: Maybe this doesn't need to be part of this PR, fewer things to argue about...

File: packages/next/next-server/server/image-optimizer.ts [hunk 2]
```diff
@@ -35,7 +35,8 @@ export async function imageOptimizer(
   res: ServerResponse,
   parsedUrl: UrlWithParsedQuery,
   nextConfig: NextConfig,
-  distDir: string
+  distDir: string,
+  imageCacheDir: string | undefined,
```
Comment (Dalimil) [inline packages/next/next-server/server/image-optimizer.ts:39]: imageCacheDir?: string

File: packages/next/next-server/server/image-optimizer.ts [hunk 3]
```diff
@@ -139,7 +140,7 @@ export async function imageOptimizer(
   }
 
   const hash = getHash([CACHE_VERSION, href, width, quality, mimeType])
-  const imagesDir = join(distDir, 'cache', 'images')
+  const imagesDir = imageCacheDir || join(distDir, 'cache', 'images')
```
Comment (Dalimil) [inline packages/next/next-server/server/image-optimizer.ts:143]: imageCacheDir ?? join (...

**Review by Dalimil — DISMISSED** [review]
This would be a really nice thing to have!

**Comment by timneutkens** [comment]
Please open a feature request first before opening a PR so that there's a check for interest before adding this feature (this is mentioned in the PR template). This is done to ensure we do not add features for a tiny percentage of applications that have to be maintained indefinitely.

**Comment by zmarty** [comment]
@timneutkens Ok, here is the RFC: https://github.com/vercel/next.js/discussions/25751

**Review by ijjk — COMMENTED** [review]
Hi, I'm gonna close this as it appears to be stale and we can continue discussion on the above thread. Note: we most likely don't want to change the cacheDir location as it being in a predictable location makes it easier for CI/deployment services to handle it.

**Pull request closed without merging**